### DISCUSSION TITLE
Skill set required to contribute to llama.cpp

### DISCUSSION THREAD

**Discussion by suri-kunal** [ggml-org/llama.cpp#9347] [General]
I am going through the roadmap of llama.cpp and I am interested in trying several items listed there. While I have been working with LLMs for a while now and have extensive understanding of their inner workings, I don't know C/C++. Could someone please let me know much C/C++/CUDA does a person need to know to start contributing to this project?

**Comment by ExtReMLapin** [comment]
Well, if you want to contribute on the C++ part, yes you need to know C++ 

duh

**Reply by suri-kunal** [reply]
I thought that too but question is about how much should I know? Do you think that https://exercism.org/tracks/cpp covers the required concepts?

**Reply by ExtReMLapin** [reply]
Even if you can cpp there are llm/gpu concepts to understand

**Reply by suri-kunal** [reply]
As I had mentioned in the question, I have extensive experience with LLMs and their inner workings. As far as GPUs are concerned, I have only worked with Pytorch which means my experience with GPUs is limited to the abstraction provided by Pytorch and I don't have experience with CUDA - I can pick it up as I go along (or am I being naive here?).
